x/claim/keeper/claim.go

```diff
@@ -0,0 +1,235 @@
+package keeper
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/arkeonetwork/arkeo/x/claim/types"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/gogo/protobuf/proto"
+)
+
+// SetClaimRecord sets a claim record for an address in store
+func (k Keeper) SetClaimRecord(ctx sdk.Context, claimRecord types.ClaimRecord) error {
+	store := ctx.KVStore(k.storeKey)
+	prefixStore := prefix.NewStore(store, chainToStorePrefix(claimRecord.Chain))
+
+	bz, err := proto.Marshal(&claimRecord)
```

Comment:
```suggestion
	bz, err := k.cdc.Marshal(&claimRecord)
```

Reply:
can you help me understand the distinction here?

Reply:
The difference is `k.cdc` will be using cosmos implementation of marshal and unmarshal , which might be protobuf or maybe not , depends on settings. use `proto.Marshal` will be always using protobuf